pkg/repository/v1_repository.go

```diff
@@ -486,6 +498,24 @@ func (r *V1Repository) FetchComponent(item *v1manifest.VersionItem) (io.Reader,
 	return checkHash(reader, item.Hashes[v1manifest.SHA256])
 }
 
+// DownloadComponent downloads the component specified by item into local file
+func (r *V1Repository) DownloadComponent(item *v1manifest.VersionItem, target string) error {
+	targetDir := filepath.Dir(target)
+	err := r.mirror.Download(item.URL, targetDir)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	reader, err := os.Open(target)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	_, err = checkHash(reader, item.Hashes[v1manifest.SHA256])
```

Comment:
I think we should remove the downloaded component if the hash is not correct to avoid attacking.

Reply:
+1